src/wazuh_modules/wm_gcp.c

```diff
@@ -0,0 +1,337 @@
+    }
+
+    if (!data->pull_on_start) {
+        time_start = time(NULL);
+
+        int status = 0;
+        time_t time_start = 0;
+        time_t time_sleep = 0;
+
+        if (data->scan_day) {
+            do {
+                status = check_day_to_scan(data->scan_day, data->scan_time);
+                if (status == 0) {
+                    time_sleep = get_time_to_hour(data->scan_time);
+                } else {
+                    wm_delay(1000); // Sleep one second to avoid an infinite loop
+                    time_sleep = get_time_to_hour("00:00");
+                }
+
+                mdebug2("Sleeping for %d seconds.", (int)time_sleep);
+                wm_delay(1000 * time_sleep);
+
+            } while (status < 0);
+
+        } else if (data->scan_wday >= 0) {
+
+            time_sleep = get_time_to_day(data->scan_wday, data->scan_time);
+            minfo("Waiting for turn to evaluate.");
+            mdebug2("Sleeping for %d seconds.", (int)time_sleep);
+            wm_delay(1000 * time_sleep);
+
+        } else if (data->scan_time) {
+
+            time_sleep = get_time_to_hour(data->scan_time);
+            minfo("Waiting for turn to evaluate.");
+            mdebug2("Sleeping for %d seconds.", (int)time_sleep);
+            wm_delay(1000 * time_sleep);
+
+        } else if (data->next_time == 0 || data->next_time > time_start) {
+
+            // On first run, take into account the interval of time specified
+            time_sleep = data->next_time == 0 ?
+                        (time_t)data->interval :
+                        data->next_time - time_start;
+
+            minfo("Waiting for turn to evaluate.");
+            mdebug2("Sleeping for %ld seconds.", (long)time_sleep);
+            wm_delay(1000 * time_sleep);
+
+        }
+
+        // On first run, take into account the interval of time specified
+        if (data->time_interval == 0) {
+            data->time_interval = time_start + data->interval;
+        }
+
+        if (data->time_interval > time_start) {
+            mtinfo(WM_GCP_LOGTAG, "Waiting for turn to start fetching.");
+            time_sleep = data->time_interval - time_start;
+            wm_delay(1000 * time_sleep);
+        }
+    }
+
+    while (1) {
+        mtdebug1(WM_GCP_LOGTAG, "Starting fetching of logs.");
+
+        // Get time and execute
+        time_start = time(NULL);
+
+        wm_gcp_run(data);
+
+        mtdebug1(WM_GCP_LOGTAG, "Fetching logs finished.");
+
+        if (data->interval) {
+            time_sleep = time(NULL) - time_start;
+
+            if ((time_t)data->interval >= time_sleep) {
+                time_sleep = data->interval - time_sleep;
+                data->time_interval = data->interval + time_start;
+            } else {
+                mtwarn(WM_GCP_LOGTAG, "Interval overtaken.");
+                time_sleep = data->time_interval = 0;
+            }
+        }
+
+        // If time_sleep=0, yield CPU
+        wm_delay(1000 * time_sleep);
```

Comment:
if there is no interval defined then time_sleep=0 and log fetching is only done once. This is currently being the default configuration.

Is this desirable?